transaction/types/src/block_version.rs

```diff
@@ -120,6 +120,12 @@ impl BlockVersion {
         self.0 >= 3
     }
 
+    /// Masked amount v2 derivation [MCIP #42](https://github.com/mobilecoinfoundation/mcips/pull/42)
+    /// are introduced in block version 3
```

Comment:
Nit: Summary first, then link to MCIP
```suggestion
    /// Masked amount V2 derivation introduced with block version 3.
    /// [MCIP #42](https://github.com/mobilecoinfoundation/mcips/pull/42)
```